src/js/sonifier.js

```diff
@@ -0,0 +1,402 @@
+                    percentage: percentage,
+                    units: units,
+                    notes: {
+                        durations: noteDurations,
+                        values: noteValues
+                    },
+                    envelope: {
+                        durations: envelopeDurations,
+                        values: envelopeValues
+                    }
+                };
+                sonificationData.push(d);
+            }
+
+        });
+
+        return sonificationData;
+    };
+
+
+    floe.chartAuthoring.sonifier.getSonificationUnits = function(value, unitDivisor) {
+        var numberDivisors = Math.floor(value / unitDivisor);
+        var numberRemainders = value % unitDivisor;
+        var divisorArray =[];
+        var remainderArray = [];
+        for(var i=0; i <numberDivisors; i++) {
+            divisorArray.push(unitDivisor);
+        }
+        for(i=0; i< numberRemainders; i++) {
+            remainderArray.push(1);
+        }
+
+        return divisorArray.concat(remainderArray);
+    };
+
+    floe.chartAuthoring.sonifier.getSonificationNoteDurations = function(units, unitDivisor, noteDurationConfig) {
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return noteDurationConfig.divisorDuration;
+            } else {
+                return noteDurationConfig.remainderDuration;
+            }
+        });
+        return durations;
+    };
+
+
+    floe.chartAuthoring.sonifier.getSonificationNoteValues = function(units, unitDivisor, noteValueConfig) {
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return noteValueConfig.divisorValue;
+            } else {
+                return noteValueConfig.remainderValue;
+            }
+        });
+        return durations;
+    };
+
+    floe.chartAuthoring.sonifier.getSonificationEnvelopeDurations = function(units, unitDivisor, envelopeDurationConfig) {
+        var durations = fluid.transform(units, function (unit) {
+            if(unit === unitDivisor) {
+                return [envelopeDurationConfig.divisorDuration,envelopeDurationConfig.divisorSilence];
+            } else {
+                return [envelopeDurationConfig.remainderDuration,envelopeDurationConfig.remainderSilence];
+            }
+        });
+
+        var durationsJoined = [];
+
+        fluid.each(durations, function(duration) {
+            durationsJoined = durationsJoined.concat(duration);
+        });
+
+        return durationsJoined;
+    };
+
+    floe.chartAuthoring.sonifier.getSonificationEnvelopeValues = function(envelopeDurations, envelopeDurationConfig, envelopeValuesConfig) {
+        var envelopeValues = fluid.transform(envelopeDurations, function(duration) {
+            if(duration === envelopeDurationConfig.divisorDuration || duration === envelopeDurationConfig.remainderDuration) {
```

Comment:
@cindyli, that approach to using a ternary to figure out which key to use is awesome. :+1: